beamer/agent/chain.py

```diff
@@ -476,6 +482,143 @@ def maybe_invalidate(claim: Claim, context: Context) -> None:
     claim.start_challenge()
 
 
+def _proof_ready_for_l1_relay(request: Request) -> bool:
+    return (
+        not request.l1_resolved.is_active
+        and request.fill_tx is not None
+        and request.fill_timestamp is not None
+    )
+
+
+def _invalidation_ready_for_l1_relay(claim: Claim) -> bool:
+    return (
+        not claim.invalidated_l1_resolved.is_active
+        and claim.invalidation_tx is not None
+        and claim.invalidation_timestamp is not None
+    )
+
+
+def _l1_resolution_threshold_reached(claim: Claim, context: Context) -> bool:
+    l1_gas_cost = 1_000_000  # TODO: Adapt to real price
+    l1_gas_price = context.web3_l1.eth.gas_price
+    l1_safety_factor = 1.25
+    limit = int(l1_gas_cost * l1_gas_price * l1_safety_factor)
+
+    # Agent is claimer
+    if claim.claimer == context.address:
+        if claim.latest_claim_made.challenger_stake_total > limit:
+            return True
+    else:
+        # Agent is challenger
+        reward = claim.get_challenger_stake(context.address)
+        last_challenger = claim.latest_claim_made.last_challenger
+        if last_challenger == context.address:
+            reward -= (
+                claim.latest_claim_made.challenger_stake_total
+                - claim.latest_claim_made.claimer_stake
+            )
+
+        if reward > limit:
+            return True
+    return False
```

Comment:
I know it#s just copy paste here but you could simplify this function by pulling out the condition check like

```py
if claim.claimer == context.address:
    reward = ...
else:
    reward = ....
return reward > limit
```

Reply:
ok done